step-controller-plugins/step-controller-plugins-timeseries/src/main/java/step/plugins/timeseries/TimeSeriesService.java

```diff
@@ -111,6 +116,8 @@ public AsyncTaskStatus<Object> rebuildTimeSeries(TimeSeriesRebuildRequest reques
                     new SearchOrder(MeasurementPlugin.BEGIN, -1), 0, 1, 0).findFirst().orElse(null);
             if (firstMeasurement != null && lastMeasurement != null) {
                 return asyncTaskManager.scheduleAsyncTask(t -> {
+                    // the flushing period can be a big value, because we will force flush every time.
+                    TimeSeriesIngestionPipeline ingestionPipeline = timeSeries.newIngestionPipeline(ingestionResolutionPeriod, 30_000);// we create a new pipeline for every migration
```

Comment:
@lucianbaciu A flushing period of 30s represents a risk of out of memory. This should be reduced to a few seconds to avoid any out of memories